dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -91,6 +88,15 @@ async def get_actor_info(self):
         return actors
 
     async def get_serve_info(self):
+        # Conditionally import serve to prevent ModuleNotFoundError from serve
+        # dependencies when only ray[default] is installed (#17712)
+        try:
+            from ray.serve.controller import SNAPSHOT_KEY as SERVE_SNAPSHOT_KEY
+            from ray.serve.constants import SERVE_CONTROLLER_NAME
+            from ray.serve.kv_store import format_key
+        except Exception:
```

Comment:
maybe make this an importerror?

Reply:
If we throw something, can we be sure to catch it on the other side? Ideally we should be able to return the core information regardless of whether there a dependency issue with serve.

Reply:
In the issue it's a ModuleNotFoundError, but I can change it to check for both ImportError and ModuleNotFoundError.

@wuisawesome Sorry, could you spell out what it means to catch it on the other side?  Also, I think this code does still return the core information if serve isn't there, it'll just have `"deployments": {}` in the snapshot.